### ISSUE TITLE
Linear MatStepper step headers do not pass aXe accessibility checks

### ISSUE THREAD

**Issue by mitchellwills** [angular/material2#10727]
#### Bug, feature request, or proposal:
When in linear mode the mat stepper headers fail aXe accessibility checks because they do not have enough contrast.

#### What is the expected behavior?
Checks do not fail

#### What is the current behavior?
The number (in the circle next to the label text) fails with:
Element has insufficient color contrast of 2.67 (foreground color: #ffffff, background color: #9e9e9e, font size: 10.5pt, font weight: normal)

The actual label fails with:
Element has insufficient color contrast of 2.67 (foreground color: #9e9e9e, background color: #ffffff, font size: 10.5pt, font weight: normal)

#### What are the steps to reproduce?
https://stackblitz.com/edit/angular-vy7pqw?file=app/stepper-overview-example.html

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
HEAD

**Comment by raygervais** [comment]
I'd like to take a look into resolving this, and will start by referencing the material guidelines for accessibly related semantics. Are you testing for WCAG AA, or AAA?

**Comment by mitchellwills** [comment]
I'm not sure, I believe we are using whatever the default is for aXe.

**Comment by raygervais** [comment]
Sorry for the delay, just wrapping up some loose ends with the semester concluding. So I did some testing while running the aXe accessibly engine (which does test for WCAG AA), and found that the color contrast ratios are closer to `2.44:1` instead of `4.5:1`. 

One way I found fixing this would be to increase the contrast (which would in turn, dark the container element. 

Just an example to open up discussion: 

## Before: 
<img width="1646" alt="screen shot 2018-04-16 at 8 45 50 pm" src="https://user-images.githubusercontent.com/14265337/38842321-47956af8-41b7-11e8-8adb-22ab405e90ad.png">

`Font Color: FFF`
`Background Color: rgba(0,0,0,.38)`

---

## After: 
<img width="996" alt="screen shot 2018-04-16 at 8 49 07 pm" src="https://user-images.githubusercontent.com/14265337/38842382-a62eeecc-41b7-11e8-873f-01592b9787ba.png">

`Font Color: FFF`
`Background Color: rgba(0, 0, 0, .54)`

---

This leads into the same treatment needed to be applied to the `mat-stepper` labels, which also would need to adjust the font color to the same background color mentioned above to meet WCAG AA Compliance. 

The final 'fix' would look as the following: 
<img width="1025" alt="screen shot 2018-04-16 at 8 52 42 pm" src="https://user-images.githubusercontent.com/14265337/38842470-23a37b66-41b8-11e8-8702-5e417194b570.png">


I'm all for the fix, but looking into the Material Spec regarding accessibly, they never claim to be WCAG AA compliant. Even the website itself when analyzed by aXe fails for a multitude of color contrast related reasons. https://material.io/guidelines/usability/accessibility.html#accessibility-color-contrast

Thoughts?

**Comment by literalpie** [comment]
I believe this problem also applies to all uses of the "secondary text" color in the [theme palette](https://github.com/angular/material2/blob/c224a389b9ec8bef3f6b17eed7d937d7a5c1f77b/src/lib/core/theming/_palette.scss#L29). We are seeing the contrast test fail with the label text of a de-selected (or read-only) input field.

The Material guidelines to recommend compliance with WCAG 2.0:
> All text should be legible and meet accessibility standards. The Web Content Accessibility Guidelines (WCAG 2.0) level AA requires a 4.5.1 color contrast between text and background for normal text, and 3:1 to large text.

[source](https://material.io/design/color/text-legibility.html#legibility-standards)

**Comment by mmalerba** [comment]
@raygervais sorry for slow response, I've been OOO for a while. I think it's more important to be accessible than exactly on spec. The change in your screenshot looks good if you want to send a PR for it
